americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+        on the plot in the upper right hand corner by default.
+    show_xgrid: bool, optional
+        Default is False. Adds vertical lines at each major x-tick.
+    show_ygrid: bool, optional
+        Default is True. Adds horizonal lines at each major y-tick.
+    title: str, optional
+        The title to be placed on the graph.
+    ylims : list
+        The limits for the y-axis.
+    ylabel : str
+        The label text for the y-axis. Every time you leave off appropriate
+        labels and units, a science grad student grading lab reports cries
+        another bitter tear into their bottle of craft beer.
+
+    Returns
+    -------
+    ax : axes
+        A matplotlib axes containing the plotted data
+
+    """
+
+    grouped = meta.groupby(group)
+
+    # Sets up the plotting order
+    if order is None:
+        order = grouped.groups.keys()
+
+    # Gets the data vectors
+    vecs = [grouped.get_group(g)[cat].values for g in order]
+
+    # Formats the axis, if not already done
+    if 'xticklabels' not in boxplot_props:
+        boxplot_props['xticklabels'] = [g.split('(')[0] for g in order]
+    if 'show_xticks' not in boxplot_props:
+        boxplot_props['show_xticks'] = False
+    if 'show_ygrid' not in boxplot_props:
+        boxplot_props['show_ygrid'] = True
+
+    # Calculates the p value
+    h, p = kruskal(*vecs)
+
+    # Sets the boxplot properties
+    ax = boxplot(vecs=vecs, ax=ax, p_value=p, **boxplot_props)
+
+    return ax
+
+
+def post_hoc_pandas(meta, group, cat, order=None, correct=None,
+    show_stats=True):
+    """Preforms an post-hoc comparison between two groups
+
+    Parameters
+    ----------
+    meta : pandas DataFrame
+        the metadata object for the data
+    group : str
+        the metadata category being interograted
+    cat : str
+        the name of the column with the result
+    order : None, list, optional
+        Default is None. The order of groups in the category.
+    correct : None, str, optional
+        Method for multiple hypothesis correction using
+        `statsmodels.sandbox.stats.multicomp.multipletests`. Methods you're
+        likely to use are `bonferroni` and `fdr_bh`.
+    show_stats : bool, optional
+        When `show_stats` is True, a summary of each group will be displayed
+        along with the p values. 
+
+    Returns
+    -------
+    post_hoc : dataframe
+        `post_hoc` summarizes the results of the post-hoc test. It includes
+        statitics about each distribution, as well as the comparison matrix
+        of p-values.
+
+    """
+
+    # Groups the data
+    grouped = meta.groupby(group)
+
+    # Gets the order
+    if order is None:
+        order = grouped.groups.keys()
+
+    # Prealocates an output frame
+    if show_stats:
+        stats = pd.DataFrame({'Counts': grouped[cat].count(),
+                              'Mean':  grouped[cat].mean(),
+                              'Stdv': grouped[cat].std(),
```

Comment:
Never used stdv in excel, but I actively try to avoid that piece of
software if possible. Flagged because I've never seen standard deviation
referred to in python (or other packages) as that. no need to change, just
curious

On Tue, Mar 3, 2015 at 4:25 PM, J W Debelius <notifications@github.com>
wrote:

> In americangut/diversity_analysis.py
> <https://github.com/biocore/American-Gut/pull/127#discussion_r25738641>:
>
> > +        of p-values.
> > +
> > +    """
> > +
> > +    # Groups the data
> > +    grouped = meta.groupby(group)
> > +
> > +    # Gets the order
> > +    if order is None:
> > +        order = grouped.groups.keys()
> > +
> > +    # Prealocates an output frame
> > +    if show_stats:
> > +        stats = pd.DataFrame({'Counts': grouped[cat].count(),
> > +                              'Mean':  grouped[cat].mean(),
> > +                              'Stdv': grouped[cat].std(),
>
> Standard devation. stdv is the function in excel.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/American-Gut/pull/127/files#r25738641>.
>